### PR TITLE
Add sequence number to API response & send center letter and digit

### PR DESCRIPTION
**Pull request by angelix**
- Add sequence number to API response (field name: "#")
- Add a checkbox for whether to send the center letter and digit.

Based on https://github.com/dicekeys/dicekeys-app-typescript/pull/200

adb_screenshot

### CODE REVIEW

**Comment by UppaJung** [comment]
Looks nice.  May not have time to review code today.

File: app/src/main/java/org/dicekeys/app/viewmodels/ApiRequestViewModel.kt [hunk 1]
```diff
@@ -2,25 +2,80 @@ package org.dicekeys.app.viewmodels
...
+    fun toggleHideFaces() {
+        diceKeyRepository.toggleHideFaces()
+    }
+
+    fun sequenceUp() {
+        updateSequence((sequence.value?.toIntOrNull() ?: 1) + 1)
+    }
+
+    fun sequenceDown() {
+        updateSequence((sequence.value?.toIntOrNull() ?: 2) - 1)
```
Comment (UppaJung) [inline app/src/main/java/org/dicekeys/app/viewmodels/ApiRequestViewModel.kt:52]: will this change sequence to null if going down from 2?

File: app/src/main/java/org/dicekeys/app/viewmodels/ApiRequestViewModel.kt [hunk 2]
```diff
@@ -2,25 +2,80 @@ package org.dicekeys.app.viewmodels
...
+    }
+
+    fun updateSequence(s: Int) {
+        if (s > 0) {
+            sequence.postValue(s.toString())
```
Comment (UppaJung) [inline app/src/main/java/org/dicekeys/app/viewmodels/ApiRequestViewModel.kt:57]: Should turn 1 to null?

**Review by UppaJung — CHANGES_REQUESTED** [review]
Looks good other than ensuring that when clicking sequence number down to 1 that it becomes null instead.

**Comment by angelix** [comment]
> sequence

Fixed in c5e7a1e8f58f34f8c4743ce3fa030d7c415a1742

File: app/src/main/java/org/dicekeys/app/viewmodels/ApiRequestViewModel.kt (hunk 1 above)
Comment (angelix) [reply app/src/main/java/org/dicekeys/app/viewmodels/ApiRequestViewModel.kt:52]: Fixed in ApiRequestViewModel & RecipeBuilder to be consistent.

**Review by UppaJung — APPROVED** [review]

**Pull request merged**